R/exercise.R

```diff
@@ -248,18 +250,59 @@ evaluate_exercise <- function(exercise, envir, evaluate_global_setup = FALSE) {
   knitr::opts_chunk$set(comment = NA)
   knitr::opts_chunk$set(error = FALSE)
 
+  # helper function that unpacks knitr chunk options and
+  # returns a single character vector (e.g. "tidy=TRUE, prompt=FALSE")
+  unpack_options <- function(opts) {
+    # filter out options we don't need for the exercise.Rmd
+    opts <- opts[!(names(opts) %in% c("label", "engine"))]
+    opts <- opts[!grepl("exercise", names(opts))]
+    if (length(opts) == 0)
+      return(NULL)
+    paste0(names(opts), '=', opts)
+  }
 
+  # helper function that processes a list of raw setup chunks and
+  # returns a single character vector of knitr chunks for an Rmd file
+  get_setup_chunk_rmds <- function(chunks) {
+    if (is.null(chunks)) return()
+    setup_code_list <- chunks
+    setup_rmds <- c()
+    for (i in seq_along(setup_code_list)) {
+      code_chunk <- setup_code_list[[i]]
```

Comment:
Typically `x <- c(x, a)` is slow.  Since each iteration does not depend on the prior iteration and we want a vector of values, we can use `vapply`

```suggestion
    setup_rmds <- vapply(setup_code_list, character(1), FUN = function(code_chunk) {
```

And at the bottom of this function, just return `output_for_chunk`

Reply:
Might as well just pass in `chunks`, as `setup_code_list` is not used